Coding/fw/lib/Lawicel/Lawicel.cpp

```diff
@@ -0,0 +1,1316 @@
+/***************************************************************************************************
+  (c) NewTec GmbH 2020   -   www.newtec.de
+  $URL: https://github.com/NewTec-GmbH/esp32-can-iot $
+***************************************************************************************************/
+/**
+@addtogroup Lawicel
+@{
+@file       Lawicel.cpp
+
+Driver for Lawicel Protocol @ref Lawicel.h
+
+* @}
+***************************************************************************************************/
+/* INCLUDES ***************************************************************************************/
+#include "Lawicel.h"
+
+/* C-Interface ************************************************************************************/
+extern "C"
+{
+}
+
+/* CONSTANTS **************************************************************************************/
+const uint16_t Lawicel::MAX_TIMESTAMP = 0xEA5F;
+const String Lawicel::X_VERSION = "V0101";
+const String Lawicel::X_SERIAL_NUMBER = "NNT32";
+const char Lawicel::CR = 13;
+const char Lawicel::BELL = 7;
+
+const String Lawicel::INIT_SERIAL_BAUD = "SerialBaud";
+const String Lawicel::INIT_TIMESTAMP = "ToggleTime";
+const String Lawicel::INIT_AUTO_START = "AutoStart";
+const String Lawicel::INIT_CAN_BAUD = "CanBaud";
+const String Lawicel::INIT_FILTER_MODE = "FilterMode";
+const String Lawicel::INIT_FILTER_ACN = "FilterACn";
+const String Lawicel::INIT_FILTER_AMN = "FilterAMn";
+
+/* MACROS *****************************************************************************************/
+
+/* TYPES ******************************************************************************************/
+
+/* PROTOTYPES *************************************************************************************/
+
+/* VARIABLES **************************************************************************************/
+
+/* PUBLIC METHODES ********************************************************************************/
+
+/**************************************************************************************************/
+bool Lawicel::executeCycle()
+{
+    bool success = true;
+    m_serialReturn = "";
+    
+    if (m_autoPolling)
+    {
+        if (autopoll())
+        {
+            m_selectedSerial->print(m_serialReturn += (char)CR);
```

Comment:
you should not combine the string print with the CR addition in one line.  It negatively impacts readability.